api/src/main/resources/marquez/db/migration/V46__dataset_symlinks.sql

```diff
@@ -0,0 +1,18 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+CREATE TABLE dataset_symlinks (
+  dataset_uuid      UUID,
+  name              VARCHAR(255) NOT NULL,
+  namespace_uuid    UUID REFERENCES namespaces(uuid),
+  type              VARCHAR(64),
+  is_primary        BOOLEAN DEFAULT FALSE,
+  created_at        TIMESTAMP NOT NULL,
+  updated_at        TIMESTAMP NOT NULL,
+  UNIQUE (namespace_uuid, name)
+);
+
+INSERT INTO dataset_symlinks (dataset_uuid, name, namespace_uuid, is_primary, created_at, updated_at)
+SELECT d.uuid, d.name, d.namespace_uuid, TRUE, d.created_at, d.updated_at FROM datasets d;
+
+ALTER TABLE datasets DROP CONSTRAINT datasets_namespace_uuid_name_key;
+ALTER TABLE datasets DROP COLUMN name;
```

Comment:
Mind if we split these changes into two separate migrations?

* `V46__add_dataset_symlinks.sql`
* `V46.1__drop_name_column_in_datasets.sql`

Reply:
Do we need to drop the `name` column from `datasets`? I'm wondering, can `datasets.name` be viewed as the primary name? Or, will the `name` column in `datasets_view` (introduced in https://github.com/MarquezProject/marquez/pull/2032) be considered the dataset name (=_primary_)?

Reply:
Based on the feedback, `name` column won't be dropped to mitigate the risk of migration issues and limit unexpected behavior changes.